src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -444,51 +484,54 @@ public Response get(String json) throws Exception
 			setHeaders(responseBuilder);
 			return responseBuilder.build();
 		}
-		catch (JsonSyntaxException e)
-		{
-			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
-			return builder.addError(e.getMessage()).build();
-		}
-		catch (QueryException e)
+		catch (JsonSyntaxException|QueryException|BeanValidationException e)
 		{
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
 			return builder.addError(e.getMessage()).build();
 		}
-		catch (BeanValidationException e)
-		{
-			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
-			return builder.addErrors(e.getErrorMessages()).build();
-		}
 		catch (MaxRowKeysForQueryExceededException e) {
```

Comment:
This is a very common exception but I believe it is not working (due to my n00b java skills). But in general we need some tags that can make such exception searchable, rather than just log messages.